modules/http4s/test/src-ce3/smithy4s/compliancetests/ProtocolComplianceTest.scala

```diff
@@ -0,0 +1,148 @@
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.compliancetests
+
+import alloy.SimpleRestJson
+import cats.effect.Resource
+import org.http4s._
+import org.http4s.client.Client
+import smithy4s.{Document, Service, ShapeId}
+import smithy4s.dynamic.DynamicSchemaIndex
+import smithy4s.http4s.SimpleRestJsonBuilder
+import smithy4s.kinds.FunctorAlgebra
+import weaver._
+import cats.effect.IO
+import cats.effect.std.Env
+import smithy4s.Schema
+import smithy4s.http.PayloadError
+import smithy4s.http.CodecAPI
+import smithy4s.dynamic.model.Model
+import smithy4s.dynamic.DynamicSchemaIndex.load
+import smithy4s.schema.Schema.document
+
+/**
+  * This suite is NOT implementing MutableFSuite, and uses a higher-level interface
+  * to let weaver run it.
+  *
+  * As a result, it's likely this cannot be run via clicks in IntelliJ, because the tests
+  * are dynamically created via effects, which makes it impossible to implement `RunnableSuite`,
+  * which is required to run tests via IntelliJ.
+  */
+object ProtocolComplianceTest extends EffectSuite[IO] with BaseCatsSuite {
+
+  implicit protected def effectCompat: EffectCompat[IO] = CatsUnsafeRun
+  def getSuite: EffectSuite[IO] = this
+
+  def spec(args: List[String]): fs2.Stream[IO, TestOutcome] = {
+    fs2.Stream
+      .evals(dynamicSchemaIndexLoader.map(pizzaSpec(_)))
+      .parEvalMapUnbounded(runInWeaver)
+  }
+
+  object SimpleRestJsonIntegration extends Router[IO] with ReverseRouter[IO] {
+    type Protocol = SimpleRestJson
+    val protocolTag = alloy.SimpleRestJson
+
+    def codecs = SimpleRestJsonBuilder.codecs
+
+    def routes[Alg[_[_, _, _, _, _]]](
+        impl: FunctorAlgebra[Alg, IO]
+    )(implicit service: Service[Alg]): Resource[IO, HttpRoutes[IO]] =
+      SimpleRestJsonBuilder(service).routes(impl).resource
+
+    def reverseRoutes[Alg[_[_, _, _, _, _]]](app: HttpApp[IO])(implicit
+        service: Service[Alg]
+    ): Resource[IO, FunctorAlgebra[Alg, IO]] = {
+      import org.http4s.implicits._
+      val baseUri = uri"http://localhost/"
+
+      SimpleRestJsonBuilder(service)
+        .client(Client.fromHttpApp(app))
+        .uri(baseUri)
+        .resource
+    }
+  }
+
+  private val pizzaSpec = generateTests(
+    ShapeId("alloy.test", "PizzaAdminService")
+  )
+
+  private val path = Env
+    .make[IO]
+    .get("MODEL_DUMP")
+    .map(_.fold(sys.error(""))(fs2.io.file.Path(_)))
```

Comment:
fill in the blank here

Reply:
NB : you can use `liftTo[IO](someThrowable)` on `Option`, provided you have `import cats.syntax.all._` in scope

Reply:
done